contracts/compressor/IntentCompressor.sol

```diff
@@ -0,0 +1,336 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import {IInbox} from "../interfaces/IInbox.sol";
+import {IIntentSource} from "../interfaces/IIntentSource.sol";
+import {Intent, Call, TokenAmount, Route, Reward} from "../types/Intent.sol";
+
+struct EncodedIntent {
+    uint8 sourceChainIndex;
+    uint8 destinationChainIndex;
+    // Reward token
+    uint8 rewardTokenIndex;
+    uint48 rewardAmount;
+    // Route token
+    uint8 routeTokenIndex;
+    uint48 routeAmount;
+    // Expiry duration
+    uint24 expiryDuration;
+    // Salt
+    bytes8 salt;
+}
+
+struct EncodedFulfillment {
+    uint8 sourceChainIndex;
+    uint8 destinationChainIndex;
+    // Reward token
+    uint8 routeTokenIndex;
+    uint48 routeAmount;
+    // Prove type (INSTANT = 0, BATCH = 1)
+    uint8 proveType;
+    // Claimant
+    address claimant;
```

Comment:
This should be `address recipient;` and  `claimant` should be `msg.sender`.

Reply:
Isn't the recipient the same as claimant in this context? 🤔

Reply:
We could totally constrain the recipient to always be the `msg.sender` (actually `address(this)` since this function is intended to be used with `DELEGATECALL`), but since there's enough space, I just added it here.

Reply:
On the source msg.sender is the recipient to be on destination.
But on destination, msg.sender is the solver, so solver is trying to send all the money to themselves?
This should be `address recipient;`

Reply:
I see what you mean now. Changing claimant for recipient. 👍🏻